### PR TITLE
[IMP/REF] web, mail, base: Add 'delay' param for draggable hooks, move user settings from mail to base

### PR DESCRIPTION
**Pull request by juliusc2066**
First commit introduces a "delay" parameter used to drag elements on a long mousedown instead of a short one.

This is coupled with the "tolerance" parameter to define the area in which the mousedown is considered uninterrupted.

This is directly put to use by https://github.com/odoo/enterprise/pull/38360

Second commit moves the res.users.settings from mail to base and adds the settings to session_info and the user service from web.

opw-3181082

### CODE REVIEW

File: odoo/addons/base/models/res_users_settings.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    def set_res_users_settings(self, new_settings):
+        self.ensure_one()
+        changed_settings = {}
+        for setting in new_settings.keys():
+            if setting in self._fields and new_settings[setting] != self[setting]:
+                changed_settings[setting] = new_settings[setting]
+        self.write(changed_settings)
+        formated = self._res_users_settings_format()
+        self.env['bus.bus']._sendone(self.user_id.partner_id, 'mail.record/insert', {'res.users.settings': formated})
```
Comment (Julien00859) [inline odoo/addons/base/models/res_users_settings.py:40]: base doesn't depend on bus

**Review by Julien00859 — DISMISSED** [review]

File: addons/web/static/src/core/utils/draggable_hook_builder.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
+.o_draggable {
+    touch-action: none;
```
[the remark this answers is not in the document]
Comment (Iucapad) [reply addons/web/static/src/core/utils/draggable_hook_builder.scss:2]: Maybe add this style only after the delay occurs, during the drag sequence

File: addons/web/static/tests/helpers/utils.js [hunk 3]
```diff
@@ -168,155 +180,218 @@ export function findElement(el, selector) {
...
+/**
+ * @template {EventType} T
+ * @param {Element} el
+ * @param {string | null | undefined | false} selector
+ * @param {T} eventType
+ * @param {EventInit} [eventInit]
+ * @param {TriggerEventOptions} [options]
+ * @returns {GlobalEventHandlersEventMap[T] | Promise<GlobalEventHandlersEventMap[T]>}
+ */
+export function triggerEvent(el, selector, eventType, eventInit, options) {
```
Comment (mcm-odoo) [inline addons/web/static/tests/helpers/utils.js:339]: we could prevent the use of `?.` if we default `options` with `options = {}`

**Comment by mcm-odoo** [comment]
@robodoo r+

**Comment by mcm-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**